File: app/src/main/java/info/nightscout/androidaps/plugins/pump/insight/database/InsightPumpID.java
package info.nightscout.androidaps.plugins.pump.insight.database;

import com.j256.ormlite.field.DatabaseField;
import com.j256.ormlite.table.DatabaseTable;

import info.nightscout.androidaps.db.DatabaseHelper;

@DatabaseTable(tableName = DatabaseHelper.DATABASE_INSIGHT_PUMP_IDS)
public class InsightPumpID {

    @DatabaseField(generatedId = true)
    public Long id;

    @DatabaseField
    public long timestamp;

    @DatabaseField
    public String eventType;

    @DatabaseField
    public String pumpSerial;

    @DatabaseField
    public long eventID;

}

File: insight/src/main/java/info/nightscout/androidaps/insight/database/InsightPumpID.kt
package info.nightscout.androidaps.insight.database

import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(tableName = DATABASE_INSIGHT_PUMP_IDS,
    indices = [Index("timestamp")])
data class InsightPumpID(
    var timestamp: Long,
    var eventType: String? = null,
    var pumpSerial: String? = "None",
    @PrimaryKey
    var eventID: Long
) {
    fun getId(): Long = eventID
}